src/marqo/s2_inference/processing/image.py

```diff
@@ -306,143 +147,279 @@ def process(self):
         self.bboxes_orig = [rescale_box(bb, self.size, self.original_size) for bb in self.bboxes]
 
 
-class PatchifyPytorch:
-    """class to do the patching
+class PatchifyModel:
+    """class to do the patching. this is the base class for model based chunking
     """
-    def __init__(self, device='cpu', size=(240, 240), nms=True, filter_bb=True):
-
+    def __init__(self, device: str = 'cpu', size: Tuple = (224, 224), min_area: float = 60*60, 
+                nms: bool = True, replace_small: bool = True, top_k: int = 10, 
+                filter_bb: bool = True, min_area_replace: float = 60*60, **kwargs):
+        """_summary_
+
+        Args:
+            device (str, optional): _description_. Defaults to 'cpu'.
+            size (Tuple, optional): _description_. Defaults to (224, 224).
+            min_area (float, optional): _description_. Defaults to 60*60.
+            nms (bool, optional): _description_. Defaults to True.
+            replace_small (bool, optional): _description_. Defaults to True.
+            top_k (int, optional): _description_. Defaults to 10.
+            filter_bb (bool, optional): _description_. Defaults to True.
+            min_area_replace (float, optional): _description_. Defaults to 60*60.
+        """
```

Comment:
Would be more readable if each arg's role is explained (how each arg affects the function's behaviour)